various-receipes/3rd-party-services-isolation/test/isolation.test.js

```diff
@@ -0,0 +1,110 @@
+const request = require("supertest");
+const sinon = require("sinon");
+const nock = require("nock");
+const {
+  initializeWebServer,
+  stopWebServer
+} = require("../../../example-application/api-under-test");
+const mailer = require("../../../example-application/libraries/mailer");
+const OrderRepository = require("../../../example-application/data-access/order-repository");
+
+let expressApp;
+let sinonSandbox;
+
+beforeAll(async (done) => {
+  // ️️️✅ Best Practice: Isolate the service under test by intercepting requests to 3rd party services 
+  nock("http://localhost/user/").get(`/1`).reply(200, {
+    id: 1,
+    name: "John",
+  }).persist();
+
+  // ️️️✅ Best Practice: Place the backend under test within the same process
+  expressApp = await initializeWebServer();
+
+  // ️️️✅ Best Practice: use a sandbox for test doubles for proper clean-up between tests
+  sinonSandbox = sinon.createSandbox();
+
+  done();
+});
+
+afterAll(async (done) => {
+  // ️️️✅ Best Practice: Clean-up resources after each run
+  await stopWebServer();
+  done();
+});
+
+beforeEach(() => {
+  if (sinonSandbox) {
```

Comment:
Should we clean nock after each call? 

Consider adding an example where there is a default answer to something (i.e. all users request) and then some specific test should override it. This triggers a challenge as the change affect the next test and might make it fail- There comes the need to cleanup after each test.